plugins/source/aws/resources/services/s3/buckets.go

```diff
@@ -52,41 +52,10 @@ func fetchS3Buckets(ctx context.Context, meta schema.ClientMeta, _ *schema.Resou
 	if err != nil {
 		return err
 	}
-
-	var wg sync.WaitGroup
-	buckets := make(chan types.Bucket)
-	errs := make(chan error)
-	for i := 0; i < fetchS3BucketsPoolSize; i++ {
-		wg.Add(1)
-		go fetchS3BucketsWorker(ctx, meta, buckets, errs, res, &wg)
-	}
-	go func() {
-		defer close(buckets)
-		for _, bucket := range response.Buckets {
-			select {
-			case <-ctx.Done():
-				return
-			case buckets <- bucket:
-			}
-		}
-	}()
-	done := make(chan struct{})
-	go func() {
-		for err = range errs {
-			cl.Logger().Err(err).Msg("failed to fetch s3 bucket")
-		}
-		close(done)
-	}()
-	wg.Wait()
```

Comment:
Using `Resolver` and `PreResourceResolver` instead of custom channels and go routines